python/cudf/cudf/dataframe/dataframe.py

```diff
@@ -201,7 +201,7 @@ def dtypes(self):
     def shape(self):
         """Returns a tuple representing the dimensionality of the DataFrame.
         """
-        return len(self), len(self._cols)
+        return len(self._index), len(self._cols)
```

Comment:
Any reason we need to use the length of the index instead of length of self? I imagine length of self checks length of the index?

Reply:
At least in the context of MultiIndexes, it is possible to have an index with non-empty _source_data attached to an empty dataframe. Pandas reports the length of the index as the length of the dataframe, not the length of the dataframe itself. This change doesn't appear to negatively affect any of the rest of cudf and passes a few more multiindex tests.

Reply:
I guess I would expect `__len__` of dataframe to return the `len(self.index)` so you could call `len(self)` here instead, but it's nitpicking so don't worry about it.

Reply:
My assumption looks to be mistaken 😄 closing.